server/odc-service/src/main/java/com/oceanbase/odc/service/task/schedule/MonitorExecutorStatusRateLimiter.java

```diff
@@ -34,24 +38,52 @@
 public class MonitorExecutorStatusRateLimiter implements StartJobRateLimiter {
 
     private final Supplier<TaskFrameworkProperties> taskFrameworkProperties;
+    private final TaskFrameworkService taskFrameworkService;
+    private long runningJobCountLimit;
+    private long totalPhysicMemory;
 
-    public MonitorExecutorStatusRateLimiter(Supplier<TaskFrameworkProperties> taskFrameworkProperties) {
+    public MonitorExecutorStatusRateLimiter(Supplier<TaskFrameworkProperties> taskFrameworkProperties,
+            TaskFrameworkService taskFrameworkService) {
         this.taskFrameworkProperties = taskFrameworkProperties;
+        this.taskFrameworkService = taskFrameworkService;
+        if (taskFrameworkProperties.get().getRunMode().isProcess() && SystemUtils.isOnLinux()) {
+            this.totalPhysicMemory =
+                    SystemUtils.getSystemTotalPhysicalMemory().convert(BinarySizeUnit.MB).getSizeDigit();
+            this.runningJobCountLimit = calculateRunningJobCountLimit();
+        }
     }
 
     @Override
     public boolean tryAcquire() {
         if (taskFrameworkProperties.get().getRunMode().isProcess() && SystemUtils.isOnLinux()) {
-            long systemFreeMemory = SystemUtils.getSystemFreeMemory().convert(BinarySizeUnit.MB).getSizeDigit();
-            int startNewProcessMemoryMinSize =
-                    taskFrameworkProperties.get().getStartNewProcessMemoryMinSizeInMilliBytes();
-            if (systemFreeMemory * 0.5 <= startNewProcessMemoryMinSize) {
-                log.warn("Current free memory lack, systemFreeMemory={}, startNewProcessMemoryMinSize={}",
-                        systemFreeMemory, startNewProcessMemoryMinSize);
+            long count = taskFrameworkService.countRunningJobs(TaskRunMode.PROCESS);
+            if (count >= runningJobCountLimit && log.isDebugEnabled()) {
```

Comment:
log level should not affect the rate limiter's behavior

Reply:
done